cfg/builder/builder_walk.cc

```diff
@@ -579,10 +579,9 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                     argLocs.emplace_back(exp.loc());
                 }
 
-                const auto kwEnd = s.numKwArgs();
-                for (auto argIdx = 0; argIdx < kwEnd; ++argIdx) {
-                    auto &key = s.getKwKey(argIdx);
-                    auto &val = s.getKwValue(argIdx);
+                for (auto pair : s.kwArgPairs()) {
```

Comment:
Not saying we should do this, but what operators need to be implemented in order to support structured binding?

https://en.cppreference.com/w/cpp/language/structured_binding#Case_2:_binding_a_type_implementing_the_tuple_operations

update: tried to answer my own question, looks like you can define "the tuple operations" and then it will work. neat.